### PR TITLE
Implement the data import cron feature (golden images)

### PR DESCRIPTION
**Pull request by nunnatsa**
The golden images feature provides root disk images for commonly used operating systems as well as continually providing updated versions of those images.

This PR implements the HCO part of this feature. HCO reads the list of the the golden images from a local file, on boot-up time, and populates the SSP CR with these images. HCO also support a new API to add custom images. If custom images added to HyperConverged CR, HCO will add them to the list in the SSP CR.

More details are here: https://github.com/kubevirt/community/blob/master/design-proposals/golden-image-delivery-and-update-pipeline.md

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
* Support the golden images feature
* Add new API to add custom golden images
```

### CODE REVIEW

File: pkg/apis/hco/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -37,7 +39,7 @@ type HyperConvergedSpec struct {
 
 	// featureGates is a map of feature gate flags. Setting a flag to `true` will enable
 	// the feature. Setting `false` or removing the feature gate, disables the feature.
-	// +kubebuilder:default={"withHostPassthroughCPU": false, "sriovLiveMigration": true}
+	// +kubebuilder:default={"withHostPassthroughCPU": false, "sriovLiveMigration": true, "dataImportCron": false}
```
Comment (tiraboschi) [inline pkg/apis/hco/v1beta1/hyperconverged_types.go:42]:
```suggestion
	// +kubebuilder:default={"withHostPassthroughCPU": false, "sriovLiveMigration": true, "commonDataImportCronEnabled": false}
```

**Comment by nunnatsa** [comment]
closing and reopening. too many ci changes. need some kind of reset.

**Comment by tiraboschi** [comment]
Ignoring slightly decreased coverage mainly due to error handling:
/override coverage/coveralls

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**